### PR TITLE
Dimensional_oM: Adding IElement3D

### LINKED ISSUE #1018: Dimensional_oM: Creation of IElement3D interface
#### Description:

Much like the other spatial IElement interfaces. This will be used to define objects bounded by IElements of (X-1)D. In this case a list of IElement2D 
Will be useful for objects like `Cells`/`Spaces` - defined by bounding `Faces`/`Panels` for instance.

### PR DESCRIPTION
**Pull request by al-fisher**
### Issues addressed by this PR

Closes #1018

### CODE REVIEW

**Review by jamesramsden-bh — DISMISSED** [review]
Looks very excellent :)

File: Dimensional_oM/IElement3D.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using System.ComponentModel;
+
+namespace BH.oM.Dimensional
+{
+    [Description("Enables geometrical operations to be performed on a Solid based spatial element, whilst preserving all other object properties as unchanged.")]
+    internal class IElement3D
```
Comment (FraserGreenroyd) [inline Dimensional_oM/IElement3D.cs:29]:
```suggestion
    public interface IElement3D : IElement
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
The change requested allows the interface to be in line with existing `IElementXD` elements.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by al-fisher** [comment]
Gosh! Yes! 😕 
Thanks @FraserGreenroyd good fix!
That's why implementing an interface at the same time as discussing surrogate models is not a good idea! 😆

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check serialisation

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Review by FraserGreenroyd — APPROVED** [review]
Code review now looks good, happy to deploy.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**